org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMaven.java

```diff
@@ -106,7 +106,10 @@ boolean isUnavailable(String groupId, String artifactId, String version, String
    * @since 1.4
    * @see {@link #readMavenProjects(File, ProjectBuildingRequest)} to group requests and improve performance (RAM and
    *      CPU)
+   * @deprecated this method should never have been API and is prone to errors, if you still need this method please
+   *             contact the m2e team to provide better alternatives for your use-case
    */
+  @Deprecated(forRemoval = true)
   MavenExecutionResult readMavenProject(File pomFile, ProjectBuildingRequest configuration) throws CoreException;
```

Comment:
Shouldn't this changed to be a default implementation that calls the implementation from the IMavenToolBox with a custom/standalone (just how it is done a.t.m) component-lookup?

Reply:
Actually I don't want to change the deprecated stuff as in the past ti was much more work to delegate "right" to the new impl than just keep the old impl alone.

Reply:
Understand. Keeping the old one as it is also prevents from accidental behavior changes due to differences in the new method.

Reply:
Yep, especially if we want to change the new method in an incompatible way this is just too much hassle. This way we simply can delete the method at m2e 4.0 and until then no much need to touch it anyways.